frontend/src/components/PostDialog.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Dialog, DialogTitle, DialogContent, Button, List, ListItem, Typography } from '@mui/material';
+
+const PostDialog = ({ open, onClose, project }) => {
+  if (!project) return null;
+
+  const {
+    project_name,
+    description,
+    desired_qualifications,
+    umbrella_topics = [],
+    research_periods = [],
+    is_active,
+    majors = [],
+    faculty = {},
+  } = project;
+
+  return (
+    <Dialog open={open} onClose={onClose} maxWidth="md" fullWidth>
```

Comment:
remove the reference to the open prop here. Instead we can do open={{!!post}} (two exclamation marks). This is a dynamic way to show the Dialog if post is not null. It's easier to set this dynamically inside the component itself instead of having to write logic somewhere else that says whether we should say open is true or false.